### PR TITLE
Use Jdk8Module to parse configs

### PR DESCRIPTION
**Pull request by hsaraogi**
From @jeremyk-91's suggestion

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1362)

### CODE REVIEW

File: atlasdb-timelock-server/build.gradle [hunk 1]
```diff
@@ -24,6 +24,7 @@ dependencies {
     compile group: 'io.atomix', name: 'atomix'
     compile group: 'io.atomix.catalyst', name: 'catalyst-netty'
     compile group: 'io.dropwizard', name: 'dropwizard-core'
+    compile "com.fasterxml.jackson.datatype:jackson-datatype-jdk8"
```
Comment (jeremyk-91) [inline atlasdb-timelock-server/build.gradle]: Let's try to be consistent with specifying the sections (`group` and `name`) out fully.

nit: Also could we move this to the top, to maintain alphabetical order?

File: atlasdb-timelock-server/var/conf/timelock.yml [hunk 2]
```diff
@@ -6,3 +6,9 @@ cluster:
   localServer: localhost:8700
   servers:
     - localhost:8700
+
+atomix:
+  security:
```
Comment (jeremyk-91) [inline atlasdb-timelock-server/var/conf/timelock.yml]: Thanks for the example. I'm a little concerned, though - if you don't have a `truststore.jks` or `keystore.jks` in the module directory, will this prevent the server from starting up? (If it doesn't block that, then this is good.)

Comment (jeremyk-91) [inline atlasdb-timelock-server/var/conf/timelock.yml]: Cool, thanks

File: atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServer.java [hunk 3]
```diff
@@ -56,6 +58,12 @@ public static void main(String[] args) throws Exception {
     }
 
     @Override
+    public void initialize(Bootstrap<TimeLockServerConfiguration> bootstrap) {
+        super.initialize(bootstrap);
```
Comment (SerialVelocity) [inline atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/TimeLockServer.java]: No-one seems to call `super.initialize` in other repos as the `Application.initialize` is known to be empty.

**Pull request merged**